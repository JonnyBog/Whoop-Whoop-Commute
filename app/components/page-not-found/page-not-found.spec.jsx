import React from 'react';
import { shallow } from 'enzyme';

import PageNotFound from 'components/page-not-found/page-not-found';

describe('Components', () => {
  describe('PageNotFound', () => {
    it('should export', () => {
      expect(PageNotFound).toBeDefined();
    });

    /**
     * createShallowWrapper
     * @returns {JSX} shallow component
     */
    function createShallowWrapper () {
      const history = {};

      return shallow(
        <PageNotFound history={history} />
      );
    }

    describe('Appearance', () => {
      it('should render copy', () => {
        const wrapper = createShallowWrapper();
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
