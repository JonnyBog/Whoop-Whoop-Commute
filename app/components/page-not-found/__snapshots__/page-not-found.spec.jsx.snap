// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components PageNotFound Appearance should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageNotFound
    history={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
        >
          <title>
            Whoop Whoop Commute - Page not found
          </title>
        </HelmetWrapper>,
        <GridContainer>
          <GridItem
            width={
              Array [
                1,
                1,
                1,
              ]
            }
          >
            <GridItem
              px={
                Array [
                  0,
                  0,
                  0,
                ]
              }
              width={
                Array [
                  1,
                  1,
                  0.5,
                ]
              }
            >
              <p>
                Page not found!
                <br />
                <br />
                <Link
                  replace={false}
                  to="/"
                >
                  Take me Home
                </Link>
              </p>
            </GridItem>
          </GridItem>
        </GridContainer>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <title>
            Whoop Whoop Commute - Page not found
          </title>,
          "defer": true,
          "encodeSpecialCharacters": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Whoop Whoop Commute - Page not found",
          },
          "ref": null,
          "rendered": "Whoop Whoop Commute - Page not found",
          "type": "title",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <GridItem
            width={
              Array [
                1,
                1,
                1,
              ]
            }
          >
            <GridItem
              px={
                Array [
                  0,
                  0,
                  0,
                ]
              }
              width={
                Array [
                  1,
                  1,
                  0.5,
                ]
              }
            >
              <p>
                Page not found!
                <br />
                <br />
                <Link
                  replace={false}
                  to="/"
                >
                  Take me Home
                </Link>
              </p>
            </GridItem>
          </GridItem>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GridItem
              px={
                Array [
                  0,
                  0,
                  0,
                ]
              }
              width={
                Array [
                  1,
                  1,
                  0.5,
                ]
              }
            >
              <p>
                Page not found!
                <br />
                <br />
                <Link
                  replace={false}
                  to="/"
                >
                  Take me Home
                </Link>
              </p>
            </GridItem>,
            "width": Array [
              1,
              1,
              1,
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <p>
                Page not found!
                <br />
                <br />
                <Link
                  replace={false}
                  to="/"
                >
                  Take me Home
                </Link>
              </p>,
              "px": Array [
                0,
                0,
                0,
              ],
              "width": Array [
                1,
                1,
                0.5,
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Page not found!",
                  <br />,
                  <br />,
                  <Link
                    replace={false}
                    to="/"
                  >
                    Take me Home
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Page not found!",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Take me Home",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Take me Home",
                  "type": [Function],
                },
              ],
              "type": "p",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
          >
            <title>
              Whoop Whoop Commute - Page not found
            </title>
          </HelmetWrapper>,
          <GridContainer>
            <GridItem
              width={
                Array [
                  1,
                  1,
                  1,
                ]
              }
            >
              <GridItem
                px={
                  Array [
                    0,
                    0,
                    0,
                  ]
                }
                width={
                  Array [
                    1,
                    1,
                    0.5,
                  ]
                }
              >
                <p>
                  Page not found!
                  <br />
                  <br />
                  <Link
                    replace={false}
                    to="/"
                  >
                    Take me Home
                  </Link>
                </p>
              </GridItem>
            </GridItem>
          </GridContainer>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <title>
              Whoop Whoop Commute - Page not found
            </title>,
            "defer": true,
            "encodeSpecialCharacters": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Whoop Whoop Commute - Page not found",
            },
            "ref": null,
            "rendered": "Whoop Whoop Commute - Page not found",
            "type": "title",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GridItem
              width={
                Array [
                  1,
                  1,
                  1,
                ]
              }
            >
              <GridItem
                px={
                  Array [
                    0,
                    0,
                    0,
                  ]
                }
                width={
                  Array [
                    1,
                    1,
                    0.5,
                  ]
                }
              >
                <p>
                  Page not found!
                  <br />
                  <br />
                  <Link
                    replace={false}
                    to="/"
                  >
                    Take me Home
                  </Link>
                </p>
              </GridItem>
            </GridItem>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <GridItem
                px={
                  Array [
                    0,
                    0,
                    0,
                  ]
                }
                width={
                  Array [
                    1,
                    1,
                    0.5,
                  ]
                }
              >
                <p>
                  Page not found!
                  <br />
                  <br />
                  <Link
                    replace={false}
                    to="/"
                  >
                    Take me Home
                  </Link>
                </p>
              </GridItem>,
              "width": Array [
                1,
                1,
                1,
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <p>
                  Page not found!
                  <br />
                  <br />
                  <Link
                    replace={false}
                    to="/"
                  >
                    Take me Home
                  </Link>
                </p>,
                "px": Array [
                  0,
                  0,
                  0,
                ],
                "width": Array [
                  1,
                  1,
                  0.5,
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Page not found!",
                    <br />,
                    <br />,
                    <Link
                      replace={false}
                      to="/"
                    >
                      Take me Home
                    </Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Page not found!",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Take me Home",
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "Take me Home",
                    "type": [Function],
                  },
                ],
                "type": "p",
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
